diff -ur libiconv-1.13.1/srclib/stdint.in.h /Users/nkeynes/src/libiconv-1.13.1/srclib/stdint.in.h
--- libiconv-1.13.1-orig/srclib/stdint.in.h	2009-06-21 21:31:08.000000000 +1000
+++ libiconv-1.13.1/srclib/stdint.in.h	2012-02-16 09:57:24.000000000 +1000
@@ -29,6 +29,13 @@
    <inttypes.h>.  */
 #define _GL_JUST_INCLUDE_SYSTEM_INTTYPES_H
 
+/* On Android (Bionic libc), <sys/types.h> includes this file before
+   having defined 'time_t'.  Therefore in this case avoid including
+   other system header files; just include the system's <stdint.h>.  */
+#if defined __ANDROID__
+# @INCLUDE_NEXT@ @NEXT_STDINT_H@
+#else
+
 /* Get those types that are already defined in other system include
    files, so that we can "#define int8_t signed char" below without
    worrying about a later system include file containing a "typedef
@@ -565,3 +572,4 @@
 
 #endif /* _GL_STDINT_H */
 #endif /* !defined _GL_STDINT_H && !defined _GL_JUST_INCLUDE_SYSTEM_STDINT_H */
+#endif /* !__ANDROID__ */
\ No newline at end of file
